rskj-core/src/test/java/co/rsk/peg/feeperkb/FeePerKbSupportImplTest.java

```diff
@@ -0,0 +1,191 @@
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+        Coin negativeFeePerKbVote = Coin.NEGATIVE_SATOSHI;
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, negativeFeePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.NEGATIVE_FEE_VOTED.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_aboveMaxFeePerKbValue_shouldReturnExcessiveFeeVotedResponseCode() {
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+        Coin maxFeePerKb = feePerKbConstants.getMaxFeePerKb();
+        Coin excessiveFeePerKbVote = maxFeePerKb.add(Coin.SATOSHI);
+
+        Integer aboveMaxValueVotedResult = feePerKbSupport.voteFeePerKbChange(tx, excessiveFeePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.EXCESSIVE_FEE_VOTED.getCode();
+        assertEquals(expectedResult, aboveMaxValueVotedResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_unsuccessfulVote_shouldReturnUnsuccessfulFeeVotedResponseCode() {
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+
+        ABICallElection feePerKbElection = mock(ABICallElection.class);
+        when(feePerKbElection.vote(any(), any())).thenReturn(false);
+
+        AddressBasedAuthorizer authorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
+        when(storageProvider.getFeePerKbElection(authorizer)).thenReturn(feePerKbElection);
+
+        Coin feePerKbVote = Coin.valueOf(50_000L);
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, feePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.UNSUCCESSFUL_VOTE.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_successfulFeePerKbVote_shouldReturnSuccessfulFeeVotedResponseCode() {
+        SignatureCache signatureCache = mock(SignatureCache.class);
+        Transaction tx = getTransactionFromAuthorizedCaller(signatureCache);
+
+        ABICallElection feePerKbElection = mock(ABICallElection.class);
+        AddressBasedAuthorizer authorizer = feePerKbConstants.getFeePerKbChangeAuthorizer();
+        when(storageProvider.getFeePerKbElection(authorizer)).thenReturn(feePerKbElection);
+        when(feePerKbElection.vote(any(), any())).thenReturn(true);
+
+        Coin feePerKbVote = Coin.valueOf(50_000L);
+
+        Integer actualResult = feePerKbSupport.voteFeePerKbChange(tx, feePerKbVote, signatureCache);
+
+        Integer expectedResult = FeePerKbResponseCode.SUCCESSFUL_VOTE.getCode();
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void voteFeePerKbChange_successfulFeePerKbChanged_shouldReturnSuccessfulFeeVotedResponseCode() {
```

Comment:
What's the difference between this test and the one above

Reply:
This test case is related to a Fee per Kb that was successfully changed. The previous test case is a Fee per Kb successfully voted.

Reply:
the method is for voting a fee per kb change. So if you successfully changed it, the fee per kb was successfully voted -and viceversa-.

Reply:
After a discussion with @julia-zack , we agreed that `SUCCESSFUL_VOTE` is a state that might happen in 2 situations as I described above.